### ISSUE TITLE
v0.14.3 world no longer works after updating to v0.15.0

### ISSUE THREAD

**Issue by iposchalk** [PocketMine/PocketMine-MP#4216]
After updating my children's Pocketmine-MP server .phar file from "1.6dev-26_80d6f8df_API-2.0.0" (v0.14.3) to "1.6dev-27_ef8227a0_API-2.0.0" (V0.15.0), they can no longer join to their world. Their iOs Minecraft v0.15.0 app crashes. The Pocketmine-MP Server recognises them joining, but after the app crashes, the server states that they have left the game, "logged out due to timeout"

To help me work out where the problem may lie, I have created a new Pocketmine-MP Server with the exact same PHP Binarys and .phar file. The set up creates a new blank world, which works fine with the  iOs v0.15.0 app. I then swap their previously World directory and files that worked with v0.14.3 with the created one and it no longer works as per above. 

I can only assume that the new .phar file or the v0.15.0 app is incompatible with some data in World files. I do not know how to determine what is wrong. My children have done a fair bit of work in their world and I would hate for them to lose it all like what happened way back in 2014 when Minecraft PE went from v0.8.1 to v.0.9.0. There are no crash dumps. I have attached our Worlds directory for testing if possible.

Thanks in advance for looking at this issue.

[worlds.zip](https://github.com/PocketMine/PocketMine-MP/files/329042/worlds.zip)

**Comment by rudders** [comment]
Same issue but our world was pretty new so I blew it away - but definitely an issue.

**Comment by GlitchPlayer** [comment]
Try deleting the level.dat of the 0.14.3 world, then pocketmine will
generate new level.dat Don't worry, it won't corrupt your world

**Comment by iposchalk** [comment]
I deleted the level.dat file and started the server. A new level.dat file was created, but sadly the issue was not resolved. I even tried copying the level.dat file from my new working version, but the same things happens. I looked at the properties of the new level.dat file and noticed that the modified date is today, 23Jun16 but the created date is 13Jun16, if that helps and maybe confuses.

**Comment by robske110** [comment]
This is a mcpe client issue. Somewhere in your world are wrong blocks and since 0.15 the client unexecptly crashes when trying to render them. This may be fixed in 0.15.1

**Comment by iposchalk** [comment]
In the back on my mind, I thought it may be a type of block confilct issue. The nice thing would be to work out what the block type or types are, just in case I can remove them from the world in case it won't be fixed in 15.1. That's if it is a feasible thing to try and do.

**Comment by GlitchPlayer** [comment]
Might be an issue of unknown blocks

**Comment by Creeperface01** [comment]
no it still isnt fixed even in 0.15.1 :/

**Comment by iposchalk** [comment]
That's sad to hear. v0.15.1 hasn't been released in Australia yet. I'm sot sure who develops Pocketmine-MP or how to get them to hear about this issue, assuming they don't know about it. I wonder if Shoghicp knows about it all assuming he still works for Mojang.

**Comment by GlitchPlayer** [comment]
Based on genisys issue, worlds are not working because of entities

**Comment by robske110** [comment]
No, because of render errors of unknown blocks, allthough some enitites do it too.

**Comment by Driesboy** [comment]
use world edit and remove it like i have do

**Comment by iposchalk** [comment]
Hi Driesboy, would you mind giving me a bit more detail. As much as you can please.

**Comment by Driesboy** [comment]
Remove all update blocks!

**Comment by Creeperface01** [comment]
I tried to remove all update blocks but still crashing

**Comment by Driesboy** [comment]
use clear lagg to remove the unknow mobs or entities

**Comment by Creeperface01** [comment]
Hm but if entity is unknown, pocketmine doesnt spawn it

**Comment by Driesboy** [comment]
in 0.14.3 yeah
in 0.15.0 no

**Comment by Creeperface01** [comment]
But pocketmine code is still same

**Comment by Driesboy** [comment]
yes but minecraft is updated!

**Comment by Creeperface01** [comment]
Hmmm

**Comment by iposchalk** [comment]
Driesboy, obviously to me you are a guru at all this. I am just a dad who has set up a server since around 2013 and have tried to keep it all working. (apart from the v0.8.1 to v0.9.0 issue) You mentioned to use "world edit and remove it", and "clear lagg to remove the unknown mobs or entities". Here's what I assume and I hope you can correct me as well as help me please. I use PHP7.

1) "World Edit" is a plugin. If that is right, can you send me a link. I guess I have to reload V0.14.3 to use it. If that is correct, what am I looking to remove, what commands to use or is there a one off command to remove all blocks/entities to make the world eventually work.
2) "Clear Lagg" is a plugin and all the same questions as per above.

If this is all too hard to try and explain, let me know as well. I'd love to help my kids and their friends out, but if I can't, then that's life. Feel free to download the world that is attached in this thread to see if you can see my issues. Again, only do that if you are bored and would like to give it a go for me/.

**Comment by robske110** [comment]
Hmm.
Removing entites shouldn't be hard.
I may just code a plugin wich just kills every entity on chunk load instantly.
And the unknown block problem, if it's a problem is solveable too

**Comment by RocoWayne** [comment]
so... any solution so far? Maybe a plugin that fixs this problem?

**Comment by robske110** [comment]
Workin' on it

**Comment by Creeperface01** [comment]
BigSmile said, its caused by light calculating

**Comment by RocoWayne** [comment]
I use McEdit to remove flowers, pistons and chests in some of my maps. I replace them with air blocks. And now works just fine.

**Comment by ZedCee** [comment]
I've generated some fresh maps, sans Pistons, default flowers, removed chests...
Albeit I experience with no issues from generated abandoned mine shafts or any of the dungeons, but towns just will not work correctly.

I have attempted several fixes, almost nothing works short of trimming the chunks which is no good for a custom map:
1. I've deleted all entities, villagers, chests and anything that builds.
After about 30 or so blocks became corrupted with repeat logins, deleting the path block and crop blocks, the crashes subsided.  The map now with massive holes is fully accessible (the problem was in the 2/4 clockwise) and editable in MCEdit.
2. PMImporter_dev managed to convert the file to mcregion with no issues, but even with the changes Beetroot and GrassPath are replaced by update! blocks.  The map is fully accessible and nearly undamaged, but has an unknown number of update! blocks and is uneditable in MCEdit due to SkyBlocks (a light block placed in the old world format).
Blocks noticeably 'update!'ed were GrassPath and Beetroot, but all entities were left intact, and suffering no problems from chest or spawners  too.

Copy and paste blocks and schematics do not correct the issue, lading me to believe its some mysterious block id error.  For now just crossing my fingers Mojang catches the bug in the client.

**Comment by RocoWayne** [comment]
@Creeperface01  i do several tests with the light calculating and it crash again. I remove all the light sources from the map (glowstone and sea lanterns) and recalculate the lighting with MCEDit. And still dont work

**Comment by RocoWayne** [comment]
Any news?

**Comment by iposchalk** [comment]
Hi everyone. I basically gave up as I could see no light at the end of any tunnel. But, while I have been patiently looking for the latest .PHAR file (protocol 82) for V0.15.6 off the Bintray web site, which is not there as it's latest is still for protocol 81. I did yet another global search and came across the link  (https://forums.pocketmine.net/threads/version-issues.18015/) . In this link, go and get the .PHAR file from (https://www.dropbox.com/s/tbq12l954xm27z8/PocketMine-MP 0.15.6 or 0.15.4.phar?dl=0) . This .PHAR file works for V0.15.6, protocol 82. It also seems to solve my issue and have loaded my original world with all my kids and their friends buildings etc. and it does not crash. Please, those of you who had and still have the same issue, give it a try and let us all know.

**Comment by MrGenga** [comment]
Use BlockLauncher instead, because in 0.15.0 there is bug: when it receive unknown block - it crashes. In 0.14.0 or older, it replaces with UPDATE block.

**Comment by KairusDarkSeeker** [comment]
Use BlockLauncher -_-

**Comment by ZedCee** [comment]
Except iOS users can't use something like Blockchanger, making this the rather narrow minded approach.

**Comment by dktapps** [comment]
And Windows. :+1:

**Comment by ZedCee** [comment]
Which also means GearVR users as well.

**Comment by dktapps** [comment]
Cover-all so we don't spam any more: any non-Android user will find the BlockLauncher solution ineffective.

**Comment by RocoWayne** [comment]
no one has a real solution yet?

**Comment by iposchalk** [comment]
Hi RocoWayne, as I mentioned in my last comment, I found a .PHAR file made by iTX Technology LLC that now works for me and doesn't have any of the issues I was having when I opened up this forum link. It even works with V0.15.7. We use iOS devices and our sever is on a Win10 PC. Go to the links I've previously mentioned to download the .PHAR file. or, when I run the server, it says to get the latest source code from https://github.com/iTXTech/Genisys . It's a solution for us.
